src/jit.rs

```diff
@@ -114,6 +114,11 @@ impl JitProgramSections {
         }
         Ok(())
     }
+
+    pub fn mem_size(&self) -> usize {
+        self.pc_section.len() * std::mem::size_of::<u64>() +
```

Comment:
use `self.total_allocation_size` instead because that includes the page size alignment (rounding up to the next full page size).

Reply:
Also `+ std::mem::size_of::<Self>()`?

Reply:
Why do we care about that here, what memory is actually being occupied in memory, not the length of the vector?

Reply:
There is padding **allocated** beyond the length of the vector so that the page boundary is preserved.
That is important so that no one can use that left over memory with elevated privileges for anything else.

Reply:
I see, so not reflected in the vec?  Secretly overallocated vec?

Reply:
Yes, ideally we would not use `Vec<u8>` here but `Pin<Box<[u8]>>` to prevent reallocation etc.